packages/edit-post/src/components/header/template-save-button/index.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * WordPress dependencies
+ */
+import { EntitiesSavedStates, store as editorStore } from '@wordpress/editor';
+import { Button } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ActionsPanelFill } from '../../layout/actions-panel';
+import { store as editPostStore } from '../../../store';
+
+function TemplateSaveButton() {
+	const [
+		isEntitiesReviewPanelOpen,
+		setIsEntitiesReviewPanelOpen,
+	] = useState( false );
+	const { setIsEditingTemplate } = useDispatch( editPostStore );
+	const { editEntityRecord } = useDispatch( coreStore );
+	const { getTemplateInfo, getEditedEntityRecord } = useSelect(
+		( select ) => {
+			return {
+				getTemplateInfo: select( editorStore )
+					.__experimentalGetTemplateInfo,
+				getEditedEntityRecord: select( coreStore )
+					.getEditedEntityRecord,
+			};
+		},
+		[]
+	);
+	return (
+		<>
+			<Button onClick={ () => setIsEditingTemplate( false ) } isTertiary>
```

Comment:
Should it else undo all unsaved changes applied to the template?

Reply:
I'm not sure, I think it's something we should try and adapt.